src/commands/graph/graph-pull.js

```diff
@@ -0,0 +1,93 @@
+const process = require('process')
+
+const { buildSchema } = require('graphql')
+
+const {
+  ackCLISessionEvents,
+  fetchCliSessionEvents,
+  handleCliSessionEvent,
+  loadCLISession,
+  refetchAndGenerateFromOneGraph,
+} = require('../../lib/one-graph/client')
+const { getNetligraphConfig, readGraphQLSchemaFile } = require('../../lib/one-graph/netlify-graph')
+const { NETLIFYDEVERR, chalk } = require('../../utils')
+
+const graphPull = async (options, command) => {
+  const { site, state } = command.netlify
+
+  if (!site.id) {
+    console.error(
+      `${NETLIFYDEVERR} Warning: no siteId defined, unable to start Netlify Graph. To enable, run ${chalk.yellow(
+        'netlify init',
+      )} or ${chalk.yellow('netlify link')}?`,
+    )
+    process.exit(1)
+  }
+
+  const netligraphConfig = getNetligraphConfig({ command, options })
+  const netlifyToken = await command.authenticate()
+  const siteId = site.id
+  await refetchAndGenerateFromOneGraph({ netligraphConfig, netlifyToken, state, siteId })
+
+  const oneGraphSessionId = loadCLISession(state)
+
+  if (!oneGraphSessionId) {
+    console.warn('No local Netlify Graph session found, skipping command queue drain')
+    return
+  }
+
+  const schemaString = readGraphQLSchemaFile(netligraphConfig)
+
+  let schema
+
+  try {
+    schema = buildSchema(schemaString)
+  } catch (error) {
+    console.error(`${NETLIFYDEVERR} Error parsing schema: ${error.message}`)
+    process.exit(1)
+  }
+
+  if (!schema) {
+    console.error(`${NETLIFYDEVERR} Failed to fetch and update Netlify GraphQL schem`)
+    process.exit(1)
+  }
+
+  const next = await fetchCliSessionEvents({ appId: siteId, authToken: netlifyToken, sessionId: oneGraphSessionId })
+
+  if (next.errors) {
+    console.error(`${NETLIFYDEVERR} Failed to fetch Netlify Graph cli session events`, next.errors)
+    process.exit(1)
+  }
+
+  if (next.events) {
+    const ackIds = []
+    const handled = next.events.map(async (event) => {
+      await handleCliSessionEvent({ authToken: netlifyToken, event, netligraphConfig, schema, siteId: site.id })
+      ackIds.push(event.id)
```

Comment:
This is really hard to read as you map over an array to push a value out of the promise. 

I would create an array of Promises that resolves to the values instead of having void promises.

like:

```javascript
const ackIds = await Promise.all(
  next.events.map(event => 
    handleCliSessionEvent({ 
      authToken: netlifyToken, 
      event, netligraphConfig, 
      schema, 
      siteId: 
      site.id 
    }).then(({id}) => id)
  )
);
```

Would make the code easier to understand instead pushing ids and awaiting the handles.

Reply:
I've converted this into a `for` loop anyway, since I want each event to be handled before the next.